<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title></title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">aRchaic</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/kkdey/aRchaic">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">




</div>


<div class="figure">
<img src="logo_design.svg" alt="." />
<p class="caption">.</p>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p><strong>aRchaic</strong> is an R package that performs model based clustering and visualization of ancient and modern DNA samples (see <code>archaic_plot()</code> section for example visualization). This model can be used to identify distinct patterns of DNA damage and is capable of reflecting relative rates of contamination in contaminated aDNA samples.</p>
</div>
<div id="installation" class="section level2">
<h2>Installation</h2>
<p>First and foremost, the user is required to install</p>
<ul>
<li><a href="https://www.python.org/downloads/">python</a></li>
<li><a href="https://www.rstudio.com/">R/RStudio</a></li>
</ul>
<p><strong>aRchaic</strong> requires R version to be (&gt;= 3.4). If your R version is lower, please upgrade.</p>
<p>And Python packages,</p>
<ul>
<li><a href="http://pysam.readthedocs.io/en/latest/installation.html">pysam</a></li>
<li><a href="https://pythonhosted.org/pyfaidx/#installation">pyfaidx</a></li>
</ul>
<p>Upon completion of these steps, start a new R session and install the dependency R packages.</p>
<pre class="r"><code>install.packages(&quot;devtools&quot;)
devtools::install_github(&quot;kkdey/Logolas&quot;)
devtools::install_github(&quot;kkdey/CountClust&quot;)
devtools::install_github(&quot;TaddyLab/maptpx&quot;)</code></pre>
<p>Next install the <strong>aRchaic</strong> R package*</p>
<pre class="r"><code>devtools::install_github(&quot;kkdey/aRchaic&quot;)</code></pre>
<p>Load aRchaic into R</p>
<pre class="r"><code>library(aRchaic)</code></pre>
</div>
<div id="generate-archaic-input-mff-file" class="section level2">
<h2>Generate aRchaic input (MFF file)</h2>
<p>For every mismatch in a BAM file with respect to a reference genome, <strong>aRchaic</strong> records various features of the mismatch - mismatch type, flanking bases, strand break base, strand and position of mismatch from the ends of read - into a .csv file which we call the <em>Mismatch Feature Format (MFF)</em> file.</p>
<p>A typical example MFF file looks like the following</p>
<pre class="r"><code>mff &lt;- read.csv(system.file(&quot;extdata&quot;,&quot;ancients&quot;,&quot;NE2_subsampled.csv&quot;,package = &quot;aRchaic&quot;),
                header = FALSE)
head(mff)</code></pre>
<pre><code>##       V1 V2 V3 V4 V5 V6      V7
## 1 AC-&gt;TC 17 48  A  A  - 1557694
## 2 AG-&gt;AC 47  9  A  T  + 4607735
## 3 TG-&gt;AT 93  1  A  C  - 5166724
## 4 TC-&gt;TT 16 50  G  G  - 4581952
## 5 AC-&gt;TG 48 31  A  C  +  207160
## 6 GG-&gt;AG 33  5  G  G  - 2912042</code></pre>
<p>The first column represents mismatch type and flanking bases : (left flank base)(mismatch)(right flank base) &lt;-&gt; (A)(C-&gt;T)(C). The second and third columns represent the position of mismatch from 5’ and 3’ ends of reads after mapped to reference. The 4th (5th) columns represent the base 1 base upstream (downstream) from 5’ (3’) end of read respectively. The 6th column contains information of which strand the read comes from, and the 7th column is an identifier for the read containing the mismatch.</p>
<p>We provide a script to generate the MFF files from the BAM/SAM files <a href="https://github.com/kkdey/aRchaic/tree/master/bin">here</a>. This linked folder also contains a small example BAM file for testing. The simple two-line code to create the MFF file from this BAM file is as follows</p>
<pre><code>samtools index example.bam
python generate_summary_bams.py -b example.bam -f /path/to/reference/FASTA/hs37d5
.fa -o example_mff.csv</code></pre>
</div>
<div id="archaic_prepare" class="section level2">
<h2>archaic_prepare()</h2>
<p>The user is expected to organize the MFF files in seperate folders per study. Here, we provide an example with two studies, one with 5 ancient samples derived from <a href="https://www.nature.com/articles/ncomms6257">Gamba et. al. 2014</a> and a second study with 5 modern samples derived from the 1000 Genomes Project. These two folders are automatically downloaded and saved under <code>inst/extdata/moderns</code> and <code>inst/extdata/ancients</code> when aRchaic is installed.</p>
<ul>
<li><strong>Modern study</strong> :</li>
</ul>
<pre class="r"><code>moderns_dir &lt;- system.file(&quot;extdata&quot;,&quot;moderns&quot;, package = &quot;aRchaic&quot;)
list.files(moderns_dir, pattern = &quot;.csv&quot;)</code></pre>
<pre><code>## [1] &quot;NA19788_subsampled.csv&quot; &quot;NA19789_subsampled.csv&quot;
## [3] &quot;NA19792_subsampled.csv&quot; &quot;NA19794_subsampled.csv&quot;
## [5] &quot;NA19795_subsampled.csv&quot;</code></pre>
<ul>
<li><strong>Ancient study</strong> :</li>
</ul>
<pre class="r"><code>ancients_dir &lt;- system.file(&quot;extdata&quot;,&quot;ancients&quot;, package = &quot;aRchaic&quot;)
list.files(ancients_dir, pattern = &quot;.csv&quot;)</code></pre>
<pre><code>## [1] &quot;NE2_subsampled.csv&quot; &quot;NE3_subsampled.csv&quot; &quot;NE4_subsampled.csv&quot;
## [4] &quot;NE5_subsampled.csv&quot; &quot;NE6_subsampled.csv&quot;</code></pre>
<p>Next we prepare the input data to <strong>aRchaic</strong> model from the above data folders.</p>
<pre class="r"><code>out &lt;- archaic_prepare(dirs = c(moderns_dir, ancients_dir))</code></pre>
<p>The output <code>out</code> should be a list with 2 elements - corresponding to 2 study folders. Each element is a matrix with samples along the rows, coded mismatch features along columns, and a cell in the matrix record counts of the number of times a coded mismatch pattern occurs in a sample.</p>
<p>We provide <a href="https://github.com/halasadi/modern_lite_data">here</a> the modern MFF files used in Figure 2 of our <a href="https://www.ncbi.nlm.nih.gov/pubmed/30192911">paper</a> for reproducing the analysis.</p>
</div>
<div id="archaic_fit" class="section level2">
<h2>archaic_fit()</h2>
<p><strong>aRchaic</strong> fits a mixed membership model on the prepared mismatch feature data from <code>archaic_prepare()</code> that allows each sample to have partial memberships in more than one cluster where the clusters represent distinct mismatch feature profiles.</p>
<p>We first choose a directory - <code>output_dir</code> - to save the model output and visualization.</p>
<p>Next we run the <code>archaic_fit</code> function to fit the mixed membership model. We choose the number of clusters <span class="math inline">\(K=2\)</span></p>
<pre class="r"><code>output_dir &lt;- system.file(&quot;extdata&quot;,&quot;archaic_results&quot;, package = &quot;aRchaic&quot;)
model &lt;- archaic_fit(out, K = 2, output_dir = output_dir)</code></pre>
<pre><code>## The data is read as a list of matrices - processed by archaic_prepare()</code></pre>
<pre><code>## Fitting the Grade of Membership Model - independent version - due to Y. Shiraichi and M. Stephens</code></pre>
<pre><code>## 
## Estimating on a 10 samples collection.
## Fitting the 2 clusters/topics model.
## log posterior increase: 22752.2, 63.6, 22.8, 5.8, done.</code></pre>
<p>the output <code>model</code> consists of</p>
<ul>
<li><strong>omega</strong> : the grades of membership to be plotted by stacked bar chart</li>
<li><strong>theta</strong> : cluster profiles - with each column a probability distribution on coded mismatch features</li>
<li><strong>assessment</strong> : assessment of model fit : BIC and loglikelihood</li>
</ul>
</div>
<div id="archaic_plot" class="section level2">
<h2>archaic_plot()</h2>
<p>The model output, in particular <strong>omega</strong> and <strong>theta</strong> can be visualized using the <code>archaic_plot()</code> command.</p>
<pre class="r"><code>archaic_plot(model, output_dir = output_dir, background = &quot;modern&quot;)</code></pre>
<p>Once the run finishes (a few seconds), go to the output directory <code>output_dir</code>. You should see there is a <code>structure.pdf</code> file that represents the cluster memberships in <strong>omega</strong> in a stacked bar chart (see Fig 1 left).</p>
<p>Also, the <code>output_dir</code> should contain for each <span class="math inline">\(k=1,2\)</span>, a <code>logo_clus_k.pdf</code> file visualizing the mismatch profile for each cluster as in <strong>theta</strong>. If <code>background = &quot;modern&quot;</code> in <code>archaic_plot</code> code above, this reresentation will plot enrichment of the mismatch profiles with respect to a modern background.</p>
<p>The following is a summary of these visualizations</p>
<div style="width:750px; height:750px">
<div class="figure">
<img src="vignette_fig.png" />

</div>
</div>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
    This <a href="http://rmarkdown.rstudio.com">R Markdown</a> site was created with <a href="https://github.com/jdblischak/workflowr">workflowr</a>
</p>
<hr>

<!-- To enable disqus, uncomment the section below and provide your disqus_shortname -->

<!-- disqus
  <div id="disqus_thread"></div>
    <script type="text/javascript">
        /* * * CONFIGURATION VARIABLES: EDIT BEFORE PASTING INTO YOUR WEBPAGE * * */
        var disqus_shortname = 'rmarkdown'; // required: replace example with your forum shortname

        /* * * DON'T EDIT BELOW THIS LINE * * */
        (function() {
            var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
            dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
        })();
    </script>
    <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
    <a href="http://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
-->


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
