<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Model based visualization of ancient DNA damage using *aRchaic* • aRchaic</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Model based visualization of ancient DNA damage using *aRchaic*">
<meta property="og:description" content="aRchaic">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">aRchaic</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/archaic.html">Model based visualization of ancient DNA damage using *aRchaic*</a>
    </li>
    <li>
      <a href="../articles/index.html">Model based visualization of ancient DNA damage using *aRchaic*</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/kkdey/aRchaic/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Model based visualization of ancient DNA damage using <em>aRchaic</em>
</h1>
                        <h4 class="author">Kushal K Dey</h4>
            <address class="author_afil">
      Department of Statistics, University of Chicago<br><h4 class="author">Hussein Al-Asadi</h4>
            <address class="author_afil">
      Department of Ecology and Evolution, University of Chicago<br><small class="dont-index">Source: <a href="https://github.com/kkdey/aRchaic/blob/master/vignettes/index.Rmd"><code>vignettes/index.Rmd</code></a></small>
      <div class="hidden name"><code>index.Rmd</code></div>

    </address>
</address>
</div>

    
    
<div class="figure">
<img src="logo_design.svg" alt="."><p class="caption">.</p>
</div>
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p><strong>aRchaic</strong> is an R package that performs model based clustering and visualization of ancient and modern DNA samples (see <code><a href="../reference/archaic_plot.html">archaic_plot()</a></code> section for example visualization). This model can be used to identify distinct patterns of DNA damage and is capable of reflecting relative rates of contamination in contaminated aDNA samples.</p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>First and foremost, the user is required to install</p>
<ul>
<li><a href="https://www.python.org/downloads/">python</a></li>
<li><a href="https://www.rstudio.com/">R/RStudio</a></li>
</ul>
<p><strong>aRchaic</strong> requires R version to be (&gt;= 3.4). If your R version is lower, please upgrade.</p>
<p>And Python packages,</p>
<ul>
<li><a href="http://pysam.readthedocs.io/en/latest/installation.html">pysam</a></li>
<li><a href="https://pythonhosted.org/pyfaidx/#installation">pyfaidx</a></li>
</ul>
<p>Upon completion of these steps, start a new R session and install aRchaic:</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"remotes"</span>)
<span class="kw pkg">remotes</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/remotes/man/install_github.html">install_github</a></span>(<span class="st">"kkdey/aRchaic"</span>)</pre></body></html></div>
<p>Now you should be able to load aRchaic into R.</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">aRchaic</span>)</pre></body></html></div>
<p>To ensure reproducible results, we set the seed:</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="fl">1</span>)</pre></body></html></div>
</div>
<div id="generate-archaic-input-mff-file" class="section level2">
<h2 class="hasAnchor">
<a href="#generate-archaic-input-mff-file" class="anchor"></a>Generate aRchaic input (MFF file)</h2>
<p>For every mismatch in a BAM file with respect to a reference genome, <strong>aRchaic</strong> records various features of the mismatch - mismatch type, flanking bases, strand break base, strand and position of mismatch from the ends of read - into a .csv file which we call the <em>Mismatch Feature Format (MFF)</em> file.</p>
<p>A typical example MFF file looks like the following</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="no">mff</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.csv</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>,<span class="st">"ancients"</span>,<span class="st">"NE2_subsampled.csv"</span>,<span class="kw">package</span> <span class="kw">=</span> <span class="st">"aRchaic"</span>),
                <span class="kw">header</span> <span class="kw">=</span> <span class="fl">FALSE</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">mff</span>)</pre></body></html></div>
<pre><code>##       V1 V2 V3 V4 V5 V6      V7
## 1 AC-&gt;TC 17 48  A  A  - 1557694
## 2 AG-&gt;AC 47  9  A  T  + 4607735
## 3 TG-&gt;AT 93  1  A  C  - 5166724
## 4 TC-&gt;TT 16 50  G  G  - 4581952
## 5 AC-&gt;TG 48 31  A  C  +  207160
## 6 GG-&gt;AG 33  5  G  G  - 2912042</code></pre>
<p>The first column represents mismatch type and flanking bases : (left flank base)(mismatch)(right flank base) &lt;-&gt; (A)(C-&gt;T)(C). The second and third columns represent the position of mismatch from 5’ and 3’ ends of reads after mapped to reference. The 4th (5th) columns represent the base 1 base upstream (downstream) from 5’ (3’) end of read respectively. The 6th column contains information of which strand the read comes from, and the 7th column is an identifier for the read containing the mismatch.</p>
<p>We provide a script to generate the MFF files from the BAM/SAM files <a href="https://github.com/kkdey/aRchaic/tree/master/bin">here</a>. This linked folder also contains a small example BAM file for testing. The simple two-line code to create the MFF file from this BAM file is as follows</p>
<pre><code>samtools index example.bam
python generate_summary_bams.py -b example.bam -f /path/to/reference/FASTA/hs37d5
.fa -o example_mff.csv</code></pre>
</div>
<div id="archaic_prepare" class="section level2">
<h2 class="hasAnchor">
<a href="#archaic_prepare" class="anchor"></a>archaic_prepare()</h2>
<p>The user is expected to organize the MFF files in seperate folders per study. Here, we provide an example with two studies, one with 5 ancient samples derived from <a href="https://www.nature.com/articles/ncomms6257">Gamba et. al. 2014</a> and a second study with 5 modern samples derived from the 1000 Genomes Project. These two folders are automatically downloaded and saved under <code>inst/extdata/moderns</code> and <code>inst/extdata/ancients</code> when aRchaic is installed.</p>
<ul>
<li>
<strong>Modern study</strong> :</li>
</ul>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="no">moderns_dir</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>,<span class="st">"moderns"</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">"aRchaic"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span>(<span class="no">moderns_dir</span>, <span class="kw">pattern</span> <span class="kw">=</span> <span class="st">".csv"</span>)</pre></body></html></div>
<pre><code>## [1] "NA19788_subsampled.csv" "NA19789_subsampled.csv" "NA19792_subsampled.csv"
## [4] "NA19794_subsampled.csv" "NA19795_subsampled.csv"</code></pre>
<ul>
<li>
<strong>Ancient study</strong> :</li>
</ul>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="no">ancients_dir</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>,<span class="st">"ancients"</span>, <span class="kw">package</span> <span class="kw">=</span> <span class="st">"aRchaic"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span>(<span class="no">ancients_dir</span>, <span class="kw">pattern</span> <span class="kw">=</span> <span class="st">".csv"</span>)</pre></body></html></div>
<pre><code>## [1] "NE2_subsampled.csv" "NE3_subsampled.csv" "NE4_subsampled.csv"
## [4] "NE5_subsampled.csv" "NE6_subsampled.csv"</code></pre>
<p>Next we prepare the input data to <strong>aRchaic</strong> model from the above data folders.</p>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="no">out</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/archaic_prepare.html">archaic_prepare</a></span>(<span class="kw">dirs</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="no">moderns_dir</span>, <span class="no">ancients_dir</span>))</pre></body></html></div>
<p>The output <code>out</code> should be a list with 2 elements - corresponding to 2 study folders. Each element is a matrix with samples along the rows, coded mismatch features along columns, and a cell in the matrix record counts of the number of times a coded mismatch pattern occurs in a sample.</p>
<p>We provide <a href="https://github.com/halasadi/modern_lite_data">here</a> the modern MFF files used in Figure 2 of our <a href="https://www.ncbi.nlm.nih.gov/pubmed/30192911">paper</a> for reproducing the analysis.</p>
</div>
<div id="archaic_fit" class="section level2">
<h2 class="hasAnchor">
<a href="#archaic_fit" class="anchor"></a>archaic_fit()</h2>
<p><strong>aRchaic</strong> fits a mixed membership model on the prepared mismatch feature data from <code><a href="../reference/archaic_prepare.html">archaic_prepare()</a></code> that allows each sample to have partial memberships in more than one cluster where the clusters represent distinct mismatch feature profiles.</p>
<p>We first choose a directory - <code>output_dir</code> - to save the model output and visualization.</p>
<p>Next we run the <code>archaic_fit</code> function to fit the mixed membership model. We choose the number of clusters <span class="math inline">\(K=2\)</span></p>
<div class="sourceCode" id="cb12"><html><body><pre class="r"><span class="no">output_dir</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempdir</a></span>()
<span class="no">model</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/archaic_fit.html">archaic_fit</a></span>(<span class="no">out</span>, <span class="kw">K</span> <span class="kw">=</span> <span class="fl">2</span>, <span class="kw">output_dir</span> <span class="kw">=</span> <span class="no">output_dir</span>)</pre></body></html></div>
<pre><code>## The data is read as a list of matrices - processed by archaic_prepare()</code></pre>
<pre><code>## Fitting the Grade of Membership Model - independent version - due to Y. Shiraichi and M. Stephens</code></pre>
<pre><code>## 
## Estimating on a 10 samples collection.
## Fitting the 2 clusters/topics model.
## log posterior increase: 21212.8, done.</code></pre>
<p>the output <code>model</code> consists of</p>
<ul>
<li>
<strong>omega</strong> : the grades of membership to be plotted by stacked bar chart</li>
<li>
<strong>theta</strong> : cluster profiles - with each column a probability distribution on coded mismatch features</li>
<li>
<strong>assessment</strong> : assessment of model fit : BIC and loglikelihood</li>
</ul>
</div>
<div id="archaic_plot" class="section level2">
<h2 class="hasAnchor">
<a href="#archaic_plot" class="anchor"></a>archaic_plot()</h2>
<p>The model output, in particular <strong>omega</strong> and <strong>theta</strong> can be visualized using the <code><a href="../reference/archaic_plot.html">archaic_plot()</a></code> command.</p>
<div class="sourceCode" id="cb16"><html><body><pre class="r"><span class="fu"><a href="../reference/archaic_plot.html">archaic_plot</a></span>(<span class="no">model</span>, <span class="kw">output_dir</span> <span class="kw">=</span> <span class="no">output_dir</span>, <span class="kw">background</span> <span class="kw">=</span> <span class="st">"modern"</span>)</pre></body></html></div>
<p>Once the run finishes (a few seconds), go to the output directory <code>output_dir</code>. You should see there is a <code>structure.pdf</code> file that represents the cluster memberships in <strong>omega</strong> in a stacked bar chart (see Fig 1 left).</p>
<p>Also, the <code>output_dir</code> should contain for each <span class="math inline">\(k=1,2\)</span>, a <code>logo_clus_k.pdf</code> file visualizing the mismatch profile for each cluster as in <strong>theta</strong>. If <code>background = "modern"</code> in <code>archaic_plot</code> code above, this reresentation will plot enrichment of the mismatch profiles with respect to a modern background.</p>
<p>The following is a summary of these visualizations</p>
<div style="width:750px; height:750px">
<p><img src="vignette_fig.png"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Kushal Dey, Hussein Al-Asadi.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
